src/api-docs.js

```diff
@@ -3368,6 +3368,734 @@ const defaultApiDocs = {
+                required: ['blueprint_id', 'address', 'data'],
+                properties: {
+                  blueprint_id: {
+                    type: 'string',
+                    description: 'Blueprint ID of the new nano contract.'
+                  },
+                  address: {
+                    type: 'string',
+                    description: 'Address caller that will sign the nano contract creation transaction.'
+                  },
+                  data: {
+                    type: 'object',
+                    description: 'Data of the initialize method for the nano contract created.',
+                    properties: {
+                      actions: {
+                        type: 'array',
+                        items: {
+                          type: 'object',
+                          required: ['type', 'token', 'amount'],
+                          properties: {
+                            type: {
+                              type: 'string',
+                              description: 'Type of action: \'deposit\' or \'withdrawal\'.'
+                            },
+                            token: {
+                              type: 'string',
+                              description: 'Token of the action.'
+                            },
+                            amount: {
+                              type: 'integer',
+                              description: 'Amount to deposit or withdrawal.'
+                            },
+                            address: {
+                              type: 'string',
+                              description: 'Required for withdrawal, and it\'s the address to send the token to. For deposit is optional and it\'s the address to get the utxo from.'
+                            },
+                            changeAddress: {
+                              type: 'string',
+                              description: 'Address to send the change amount. Only used for deposit and it\'s optional.'
+                            },
+                          }
+                        },
+                        description: 'List of actions for the initialize method.'
+                      },
+                      args: {
+                        type: 'array',
+                        items: {
+                          oneOf: [
+                            {
+                              type: 'string',
+                            },
+                            {
+                              type: 'integer',
+                            },
+                            {
+                              type: 'number',
+                            },
+                            {
+                              type: 'boolean',
+                            },
+                          ],
+                        },
+                        description: 'List of arguments for the initialize method.'
+                      },
+                    }
```

Comment:
We could have a nanoContractData at the beggining of the file and use this on both create and execute.
This will make updating this easier in the future.

```js
const nanoContractData = {
  type: 'object',
  description: 'Data of the initialize method for the nano contract created.',
  properties: {
    actions: {
      type: 'array',
      items: {
        type: 'object',
        required: ['type', 'token', 'amount'],
        properties: {
          type: {
            type: 'string',
            description: 'Type of action: \'deposit\' or \'withdrawal\'.'
          },
          token: {
            type: 'string',
            description: 'Token of the action.'
          },
          amount: {
            type: 'integer',
            description: 'Amount to deposit or withdrawal.'
          },
          address: {
            type: 'string',
            description: 'Required for withdrawal, and it\'s the address to send the token to. For deposit is optional and it\'s the address to get the utxo from.'
          },
          changeAddress: {
            type: 'string',
            description: 'Address to send the change amount. Only used for deposit and it\'s optional.'
          },
        }
      },
      description: 'List of actions for the initialize method.'
    },
    args: {
      type: 'array',
      items: {
        oneOf: [
          {
            type: 'string',
          },
          {
            type: 'integer',
          },
          {
            type: 'number',
          },
          {
            type: 'boolean',
          },
        ],
      },
      description: 'List of arguments for the initialize method.'
    },
  }
};
```

```suggestion
                  data: nanoContractData,
```

Reply:
Much better, thanks. Done https://github.com/HathorNetwork/hathor-wallet-headless/pull/349/commits/24c0609ced2f5eb72c7beb9fb652acfa7b0303ea